core/src/main/kotlin/utils/JobMetrics.kt

```diff
@@ -192,6 +251,21 @@ class JobMetrics(private val application: Application) : MeterBinder {
             }
         }
 
+    private fun getAnalyzerJobQueueDurations(timestamp: Instant): List<Duration> =
+        transaction {
+            AnalyzerJobsTable.select(AnalyzerJobsTable.createdAt, AnalyzerJobsTable.startedAt).where {
+                    AnalyzerJobsTable.startedAt.isNotNull()
+                }.andWhere {
+                    AnalyzerJobsTable.finishedAt.isNull()
+                }.andWhere {
+                    AnalyzerJobsTable.startedAt greaterEq timestamp
+            }.mapNotNull {
+                val createdAt = it[AnalyzerJobsTable.createdAt]
+                val startedAt = it[AnalyzerJobsTable.startedAt] ?: return@mapNotNull null
+                startedAt - createdAt
+            }
+        }
```

Comment:
Should we merge this with the function above to reduce the amount of database queries? I'm a bit undecided because it would also delay the reporting of queue times, what do you think?

Reply:
Do you want to merge the functions `getAnalyzerJobDurations` and `getAnalyzerJobQueueDurations`?

Reply:
That was the idea at least.

Reply:
It makes sense to use a single query, but I'm not sure how it would look, since the `WHERE` clauses are completely different.

Reply:
I would just use `AnalyzerJobsTable.finishedAt greaterEq timestamp` and report the queue time once the jobs have finished. That's what I meant with "it would also delay the reporting of queue times".

Reply:
But it was just an idea, I would be fine with also keeping it the way it is. We should just verify that the queries use indexes, then the performance should not be an issue.

Reply:
I would keep it as it is to avoid the delay of reporting the queue durations.

Reply:
I just checked and we are not having indexes on the columns used in these queries. I think the easiest would be to add a composite index for the status and the timestamp columns (status should go first because the index can then also be used for queries filtering only by status which should also speed up the status related metrics).

Please do some testing to verify that PostgreSQL would use the index for all queries made in the duration related metrics and add the required migration.